src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarFile.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Formats.Tar
+{
+    /// <summary>
+    /// Provides static methods for creating and extracting tar archives.
+    /// </summary>
+    public static class TarFile
+    {
+        /// <summary>
+        /// Creates a tar stream that contains all the filesystem entries from the specified directory.
+        /// </summary>
+        /// <param name="sourceDirectoryName">The path of the directory to archive.</param>
+        /// <param name="destination">The destination stream the archive.</param>
+        /// <param name="includeBaseDirectory"><see langword="true"/> to include the base directory name as the first segment in all the names of the archive entries. <see langword="false"/> to exclude the base directory name from the archive entry names.</param>
+        public static void CreateFromDirectory(string sourceDirectoryName, Stream destination, bool includeBaseDirectory)
+        {
+            throw new NotImplementedException();
```

Comment:
Why are these not implemented?

Reply:
These are the stream-based methods that were suggested and then added in the API proposal review meeting. I wanted to get the rest reviewed and not block the review on implementing these. But I'll make sure to add them before merging this PR.

Reply:
Implemented.